File: common/src/main/java/com/andyanika/translator/common/models/LanguageCode.java
package com.andyanika.translator.common.models;

public enum LanguageCode {
    RU("ru"),
    UA("uk"),
    CH("zh"),
    EN("en");

    private final String value;

    LanguageCode(String value) {
        this.value = value;
    }

    public String getValue() {
        return value;
    }
}

File: common/src/main/java/com/andyanika/translator/common/models/LanguageCode.kt
package com.andyanika.translator.common.models;

import java.nio.channels.IllegalSelectorException;

public enum LanguageCode {
    RU,
    UK,
    ZH,
    EN;

    public static LanguageCode tryParse(String value, LanguageCode defaultValue) {
        if (value == null || value.isEmpty()) {
            System.err.print("LanguageCode: failed to parse: " + value);
            return defaultValue;
        }

        try {
            return valueOf(value.toUpperCase());
        } catch (IllegalSelectorException e) {
            e.printStackTrace();
            return defaultValue;
        }
    }
}